randovania/games/fusion/exporter/hint_namer.py

```diff
@@ -0,0 +1,143 @@
+    WHITE = "[COLOR=0]"
+    RED = "[COLOR=1]"
+    PINK = "[COLOR=2]"
+    YELLOW = "[COLOR=3]"
+    GREEN = "[COLOR=4]"
+    INDIGO = "[COLOR=5]"  # Poor readability
+    TEAL = "[COLOR=6]"
+
+
+def colorize_text(color: FusionColor, text: str, with_color: bool):
+    if with_color:
+        return f"{color.value}{text}{FusionColor.RESET.value}"
+    else:
+        return text
+
+
+def _area_name(region_list: RegionList, pickup_node: PickupNode, hide_region: bool) -> str:
+    area = region_list.nodes_to_area(pickup_node)
+    if hide_region:
+        return area.name
+    else:
+        return region_list.area_name(area)
+
+
+class FusionHintNamer(HintNamer):
+    location_formatters: dict[HintLocationPrecision, LocationFormatter]
+
+    def __init__(self, all_patches: dict[int, GamePatches], players_config: PlayersConfiguration):
+        patches = all_patches[players_config.player_index]
+        location_hint_template = "{determiner.title}{pickup} can be found in {node}."
+
+        self.location_formatters = {
+            HintLocationPrecision.DETAILED: TemplatedFormatter(
+                location_hint_template,
+                self,
+            ),
+            HintLocationPrecision.REGION_ONLY: TemplatedFormatter(
+                location_hint_template,
+                self,
+            ),
+            HintLocationPrecision.RELATIVE_TO_AREA: RelativeAreaFormatter(
+                patches,
+                lambda msg, with_color: colorize_text(self.color_location, msg, with_color),
+            ),
+            HintLocationPrecision.RELATIVE_TO_INDEX: RelativeItemFormatter(
+                patches,
+                lambda msg, with_color: colorize_text(self.color_location, msg, with_color),
+                players_config,
+            ),
+        }
+
+    def format_resource_is_starting(self, resource: ItemResourceInfo, with_color: bool) -> str:
+        """Used for when an item has a guaranteed hint, but is a starting item."""
+
+        return f"{colorize_text(self.color_item, resource.long_name, with_color)} has no need to be located."
+
+    def format_guaranteed_resource(
+        self,
+        resource: ItemResourceInfo,
+        player_name: str | None,
+        location: PickupLocation,
+        hide_area: bool,
+        with_color: bool,
+    ) -> str:
+        determiner = ""
+        if player_name is not None:
+            determiner = self.format_player(player_name, with_color=with_color) + "'s "
+
+        fmt = "{} is located in {}{}."
+        location_name = self.format_location(location, with_region=True, with_area=not hide_area, with_color=with_color)
+
+        return fmt.format(
+            colorize_text(self.color_item, resource.long_name, with_color),
+            determiner,
+            location_name,
+        )
+
+    def format_location_hint(self, game: RandovaniaGame, pick_hint: PickupHint, hint: Hint, with_color: bool) -> str:
+        msg = self.location_formatters[hint.precision.location].format(
```

Comment:
mypy

Reply:
not sure how to reconcile this one

Reply:
doesn't work, checked most other games and it appears they all have this same mypy error here.

Reply:
Oh, my bad lol
`hint.precision` is the one that has the `PrecisionPair | None` type. so you can:
```py
# either try this
prec = hint.precision
assert isinstance(prec, PrecisionPair)
# or try this
prec = typing.cast(PrecisionPair, hint.precision)
# end of block


msg = self.location_formatters[prec.location].format(
```